Update cylc clean docs (#403)
cylc/cylc-doc: src/user-guide/removing-workflows.rst

   :cylc:conf:`global.cylc[install][symlink dirs]`, you might want to read
   this explanation of how Cylc handles them during cleaning.

If you manually delete a run directory (e.g., using the ``rm`` command or the
file manager), only the symlinks themselves will be deleted, not the actual
targets. In contrast, ``cylc clean`` follows the symlinks and deletes the
targets.

> - It does this only for the symlinks that can be set in

Reviewer: Take or leave, I found myself re-reading this sentence, although perhaps personal preference.
```suggestion
- It only does this for the symlinks that are set in
```
